assets/prototype/application/services/entityTextFieldSearch/index.ts

```diff
@@ -0,0 +1,28 @@
+type EntityTextFieldSearchProps = {
+	entities: any[];
+	searchText: string;
+	field?: string;
+};
+
+/**
+ * searches through provided list of entities and returns an array
+ * of entities where the set field (defaults to entity name)
+ * matches the current search text (fully or partially)
+ *
+ * @param {BaseEntity[]} entities
+ * @param {string} searchText
+ * @param {string} field
+ * @return {BaseEntity[]} entities matching search criteria
+ */
+const entityTextFieldSearch = ({ entities, searchText }: EntityTextFieldSearchProps) => {
+	const isSearchable = searchText && Array.isArray(entities);
+	const searchResult = entities.filter(({ name: title }) => {
+		const textToSearch = searchText.toLowerCase();
```

Comment:
this should be moved outside of the filter callback since it would remain the same for all entities and therefore does not have to be recalculated on every iteration

Reply:
yes, I was too focused on micro-optimizations and lost the big picture :D
thx :bowing_man: